app/transcript.py

```
@@ -14,6 +17,74 @@
+
+    IP allowlist: only globally-routable unicast addresses are permitted.
+    This implicitly rejects loopback, private, link-local, multicast,
+    unspecified (0.0.0.0 / ::), and IANA-reserved ranges — catching all the
+    categories a pure denylist approach could miss.
+    """
+    import socket as _socket
+    import ipaddress as _ipaddress
+
+    _orig_getaddrinfo = _socket.getaddrinfo
+
+    def safe_getaddrinfo(host, port, family=0, type=0, proto=0, flags=0):
```

Ruff (0.15.17)

[error] Function argument `type` is shadowing a Python builtin

(A002)